### PR TITLE
add {testdown} and {covr} reports to the pkgdown site

### PR DESCRIPTION
**Pull request by Bai-Li-NOAA**
# What is the feature?
* This PR addresses issue #1015 and partially addresses issue #862. 
* It integrates {lozen}, {testdown}, and {covr} to generate bookdown-style test and coverage reports on the FIMS {pkgdown} site.

# How have you implemented the solution?
* Added a `.covignore` file to exclude `R/zzz.R` and `R/check_fims.R` from coverage reports, as these files contain initialization or developer functions.
* Added required dependencies to the `DESCRIPTION` and `devcontainer.json` files.
* Created `.github/workflows/build-pkgdown.yml` to build the {pkgdown} site with {testdown} and {covr} reports.
* Updated `.github/workflows/call-update-pkgdown.yml` for workflow testing.
* Revised test file descriptions to ensure each test has a one-line summary placed directly above its corresponding `expect_*()` call.
* Skipped `tests/testthat/test-use-test-template.R` when generating {testdown} reports to prevent errors.

# Does the PR impact any other area of the project, maybe another repo?
* Once merged into main, and if the implementation works as expected, we can extend the same functionality to {ghactions4r}, as mentioned in issue #862.

TODO: 
- [ ] Update `.github/workflows/call-update-pkgdown.yml` after review to enable push and PR triggers for the dev and main branches.
- [ ] Squash commits into a single commit after review.
- [ ] Open an issue in {ghactions4r} to request adding support for generating these reports on the {pkgdown} site.

### CODE REVIEW

File: .github/workflows/build-pkgdown.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            pkgdown::build_site()
+          }
+        shell: Rscript {0}
+     
+      - name: save pkgdown output
+        uses: actions/upload-artifact@v4
+        with:
+          name: pkgdown-site-with-reports
+          path: docs/
+
```
Comment (e-perl-NOAA) [inline .github/workflows/build-pkgdown.yml]: I think you need to add the line from here: https://github.com/nmfs-ost/ghactions4r/blob/fd2387e472d7f9ff48da8cccbb755bfc1aecd8f6/.github/workflows/update-pkgdown.yml#L62 to actually get the pkgdown site to deploy and update it on the gh-pages branch (https://pkgdown.r-lib.org/reference/deploy_to_branch.html). Otherwise it's just saving the outputs as an artifact. At least I think this is the case.

**Review by e-perl-NOAA — CHANGES_REQUESTED** [review]

File: tests/testthat/test-integration-caa-mle-wrappers.R [hunk 2]
```diff
@@ -47,7 +47,7 @@ test_that("deterministic run returns correct nlls", {
     em_input = em_input_list[[iter_id]]
   )
 
-  test_that("deterministic run results correct number of parameters and random effects", {
+  test_that("catch-at-age model (deterministic MLE with wrappers) returns correct number of parameters and random effects", {
     #' @description Verify the number of parameters are correct
     expect_equal(get_number_of_parameters(deterministic_age_length_comp)["fixed_effects"] |> unname(), 77)
     #' @description Verify the number of random effects are correct
```
Comment (awilnoaa) [inline tests/testthat/test-integration-caa-mle-wrappers.R]: I am not sure if this is a requirement but this description also doesn't follow the "Test that" format/wording of the other descriptions

File: tests/testthat/test-integration-caa-mle.R [hunk 3]
```diff
@@ -29,8 +29,7 @@ result <- setup_and_run_FIMS_without_wrappers(
 )
 
 ## IO correctness ----
-test_that("deterministic run works with correct inputs", {
-  #' @description Compare FIMS results with model comparison project OM values.
+test_that("catch-at-age model (deterministic MLE without wrappers) works with correct inputs", {
   verify_fims_deterministic(
```
Comment (awilnoaa) [inline tests/testthat/test-integration-caa-mle.R]: A few of the verify_fims and validate_fims tests in this file do not have a description

File: tests/testthat/test-integration-fims-estimation-random-effects-with-wrappers.R [hunk 4]
```diff
@@ -82,11 +88,13 @@ test_that("deterministic test of fims with recruitment re", {
...
   # loop over years to compare recruitment by year
   for (i in 1:om_input_list[[iter_id]][["nyr"]]) {
+    #' @description Test that the recruitment values from FIMS match the "true" values from the operating model.
     expect_equal(fims_naa[i, 1], om_output_list[[iter_id]][["N.age"]][i, 1])
   }
 
   # confirm that recruitment matches the numbers in the first age
   # by comparing to fims_naa (what's reported from FIMS)
+  #' @description Test that the recruitment values from FIMS match the numbers-at-age for age 1.
   expect_equal(
```
Comment (awilnoaa) [inline tests/testthat/test-integration-fims-estimation-random-effects-with-wrappers.R]: In the "Expectation" column of the table, only a portion of the code for this test appears. In the table it is only "expect_equal("

File: tests/testthat/test-integration-fims-estimation-random-effects-with-wrappers.R [hunk 5]
```diff
@@ -156,6 +177,7 @@ test_that("deterministic test of fims with recruitment re", {
     fims_object_are[i] <- abs(fims_index[[2]][i] - em_input_list[[iter_id]][["surveyB.obs"]][["survey1"]][i]) / em_input_list[[iter_id]][["surveyB.obs"]][["survey1"]][i]
   }
   # Expect 95% of relative error to be within 2*cv
+  #' @description Test that 95% of the relative errors between the expected survey biomass from FIMS and the observed survey biomass are within 2 times the coefficient of variation.
   expect_lte(
```
Comment (awilnoaa) [inline tests/testthat/test-integration-fims-estimation-random-effects-with-wrappers.R]: This code for this is also cut off in the table and appears as "expect_lte("

File: tests/testthat/test-integration-fims-estimation-random-effects-without-wrappers.R [hunk 6]
```diff
@@ -39,7 +38,7 @@ test_that("deterministic run works with correct inputs", {
 })
 
 test_that("deterministic run returns correct nlls", {
-  #' Compare FIMS NLLs with model comparison project "true" NLLs
+  # Compare FIMS NLLs with model comparison project "true" NLLs
   verify_fims_nll(
```
Comment (awilnoaa) [inline tests/testthat/test-integration-fims-estimation-random-effects-without-wrappers.R]: No description appearing for the "verify_fims" tests

File: tests/testthat/test-parallel-caa-mle-wrappers.R [hunk 7]
```diff
@@ -98,8 +98,7 @@ test_that("Run FIMS in parallel using {snowfall}", {
 
   # Comparison of results:
   # Verify that SSB values from both runs are equivalent.
-  #' @description Test that SSB values from parallel runs equal those from
-  #' serial runs.
+  #' @description Test that SSB values from parallel runs equal those from serial runs.
   expect_setequal(
```
Comment (awilnoaa) [inline tests/testthat/test-parallel-caa-mle-wrappers.R]: The code for a few of these seems cut off. Ex. "expect_setequal(". Maybe this happens because they are skipped?

File: tests/testthat/test-use-test-template.R [hunk 8]
```diff
@@ -24,152 +35,178 @@ on.exit(
...
-    object = FIMS:::use_gtest_template(
-      name = "ClassName_FunctionName"
-    ),
-    regexp = "Invalid `name` format"
+test_that("`use_gtest_template()` handles edge cases correctly", {
+  withr::with_dir(
+    new = temp_path,
+    code = {
+      #' @description Test that `use_gtest_template()` throws an error when format of name is wrong.
+      expect_error(
```
Comment (awilnoaa) [inline tests/testthat/test-use-test-template.R]: The code for a few of these "expect_error(" are cut off in the Expectation section of the table in this file.

**Comment by Bai-Li-NOAA** [comment]
> @Bai-Li-NOAA , we talked about this on our call, but in the "Description of coverage report" , the "Hits/Line" section has a formula and example. To make it a little clearer, the multiplication order could be flipped so it matches what’s being described. It should read:
> 
> ((6 × 4) + (1 × 1) + (4 × 10)) / 11 = 6
> 
> That way it’s clear to the reader that the first 6 lines are each tested 4 times, the 7th once, and the last 4 lines 10 times.

@awilnoaa Thanks for the suggestion! The `Description of coverage report` actually comes from a different package `{lozen}`. I’ve submitted a [PR](https://github.com/ThinkR-open/lozen/pull/76) to their repo, and hopefully it will be addressed.

File: .github/workflows/build-pkgdown.yml (hunk 1 above)
Comment (Bai-Li-NOAA) [reply .github/workflows/build-pkgdown.yml]: Yes, and yes. I had the deploy code commented out, but I’ll uncomment it now.

File: tests/testthat/test-integration-caa-mle-wrappers.R [hunk 9]
```diff
@@ -20,7 +20,7 @@ load(test_path("fixtures", "integration_test_data.RData"))
 iter_id <- 1
 
 ## IO correctness ----
-test_that("deterministic run works with correct inputs", {
+test_that("catch-at-age model (deterministic MLE with wrappers) works with correct inputs", {
```
[the remark this answers is not in the document]
Comment (Bai-Li-NOAA) [reply tests/testthat/test-integration-caa-mle-wrappers.R]: Done, added a description on line 27.

File: tests/testthat/test-integration-caa-mle-wrappers.R [hunk 10]
```diff
@@ -65,7 +65,7 @@ test_that("deterministic run returns correct nlls", {
 ## Setup ----
 
 ## IO correctness ----
-test_that("estimation test with age and length comp using wrappers", {
+test_that("catch-at-age model (estimation MLE with wrappers) works with age and length comp", {
```
[the remark this answers is not in the document]
Comment (Bai-Li-NOAA) [reply tests/testthat/test-integration-caa-mle-wrappers.R]: Done. Added descriptions.

File: tests/testthat/test-integration-fims-estimation-random-effects-without-wrappers.R (hunk 6 above)
Comment (Bai-Li-NOAA) [reply tests/testthat/test-integration-fims-estimation-random-effects-without-wrappers.R]: Done. Added descriptions.

File: tests/testthat/test-integration-fims-estimation-random-effects-with-wrappers.R (hunk 5 above)
Comment (Bai-Li-NOAA) [reply tests/testthat/test-integration-fims-estimation-random-effects-with-wrappers.R]: Done. I shortened some expect_*() calls to display the code properly.

File: tests/testthat/test-rcpp-distribution.R [hunk 11]
```diff
@@ -458,15 +444,13 @@ test_that("rcpp distribution returns correct error messages", {
     \(x) dlnorm_$log_sd[x]$value <- log(1)
   )
   # TODO: skip test until dimension checking is fixed in lognormal_lpdf.hpp
+  # #' @description dlnorm should error out when there is a dimension mismatch
   # expect_error(
```
[the remark this answers is not in the document]
Comment (Bai-Li-NOAA) [reply tests/testthat/test-rcpp-distribution.R]: Done. I shortened some expect_*() calls to display the code properly.

File: tests/testthat/test-use-test-template.R (hunk 8 above)
Comment (Bai-Li-NOAA) [reply tests/testthat/test-use-test-template.R]: Done. I shortened some expect_*() calls to display the code properly.
